### ISSUE TITLE
curl on darwin is missing ipv6 support

### ISSUE THREAD

**Issue by a-n-n-a-l-e-e** [NixOS/nixpkgs#276961]
### Describe the bug
curl on darwin (x64) is not built with ipv6 support.

the ipv6 detect code in [configure.ac](https://github.com/curl/curl/blob/69b6953681d3e3172e27ef59c541d6b705ad6f9b/configure.ac#L1659C1-L1659C1) fails with a compiler error due to the function not declaring a return type.

upstream PR https://github.com/curl/curl/pull/12607 merged as https://github.com/curl/curl/commit/ae75db3527461248b0a7b1686df52200d28d83b9

### Steps To Reproduce
```console
./repl-result-bin/bin/curl -6 google.com
curl: (6) Could not resolve host: google.com
./repl-result-bin/bin/curl --version
curl 8.4.0 (x86_64-apple-darwin22.6.0) libcurl/8.4.0 OpenSSL/3.0.12 zlib/1.3 brotli/1.1.0 zstd/1.5.5 libidn2/2.3.4 libssh2/1.11.0 nghttp2/1.57.0
Release-Date: 2023-10-11
Protocols: dict file ftp ftps gopher gophers http https imap imaps mqtt pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli GSS-API HSTS HTTP2 HTTPS-proxy IDN Kerberos Largefile libz NTLM SPNEGO SSL threadsafe TLS-SRP UnixSockets zstd
```

### Notify maintainers
@lovek323

### Metadata

nixpkgs @ change d6863cbcbbb80e71cecfc03356db1cda38919523

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 21.6.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by a-n-n-a-l-e-e** [comment]
the problem is a compile error in [configure.ac](https://github.com/curl/curl/blob/69b6953681d3e3172e27ef59c541d6b705ad6f9b/configure.ac#L1659C1-L1659C1) due to the function declaring a return type.

**Comment by a-n-n-a-l-e-e** [comment]
https://github.com/curl/curl/pull/12607

**Comment by a-n-n-a-l-e-e** [comment]
fixed by #277471
